pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -157,17 +168,23 @@ func (a *Assignment) ToAPI() []kueue.PodSetAssignment {
 }
 
 // TotalRequestsFor - returns the total quota needs of the wl, taking into account the potential
-// scaling needed in case of partial admission.
+// workload slice replacement, or scaling needed in case of partial admission.
+//
+// Note: ElasticJobsViaWorkloadSlices is mutually exclusive with PartialAdmission.
 func (a *Assignment) TotalRequestsFor(wl *workload.Info) resources.FlavorResourceQuantities {
 	usage := make(resources.FlavorResourceQuantities)
 	for i, ps := range wl.TotalRequests {
-		// in case of partial admission scale down the quantity
-		aps := a.PodSets[i]
-		if aps.Count != ps.Count {
-			ps = *ps.ScaledTo(aps.Count)
+		// Either use replacement workload-slice or partial admission to scale usage.
+		if a.replaceWorkloadSlice != nil {
+			ps = *ps.ScaledTo(ps.Count - a.replaceWorkloadSlice.TotalRequests[i].Count)
+		} else {
+			aps := a.PodSets[i]
+			if aps.Count != ps.Count {
+				ps = *ps.ScaledTo(aps.Count)
+			}
 		}
```

Comment:
Instead of doing two independent branches could we just decrement the value we are scaling?

Something like:
newCount := a.PodSets[i].Count
if a.replaceWorkloadSlice != nil {
  newCount -= a.replaceWorkloadSlice.TotalRequests[i].Count
}
ps = *ps.ScaledTo(newCount)
```
wdyt?